adapters/repos/db/migrator.go

```diff
@@ -253,6 +253,63 @@ func (m *Migrator) RecalculateVectorDimensions(ctx context.Context) error {
 	return nil
 }
 
+func (m *Migrator) RecountProperties(ctx context.Context) error {
+	count := 0
+	m.logger.
+		WithField("action", "recount").
+		Info("Recounting properties, this may take a while")
+
+	clearedShard := map[string]bool{}
+
+	// Iterate over all indexes
+	for _, index := range m.db.indices {
+		// Iterate over all shards
+		err := index.IterateObjects(ctx, func(index *Index, shard *Shard, object *storobj.Object) error {
+			if _, ok := clearedShard[shard.name]; !ok {
+				shard.propLengths.Clear()
+				clearedShard[shard.name] = true
+			}
+			count = count + 1
+			props, _, err := shard.analyzeObject(object)
+			if err != nil {
+				m.logger.WithField("error", err).Error("could not analyze object")
+				return nil
+			}
+
+			if err := shard.addPropLengths(props); err != nil {
+				m.logger.WithField("error", err).Error("could not add prop lengths")
+				return nil
+			}
+
+			shard.propLengths.Flush(false)
+
+			return nil
+		})
+		if err != nil {
+			m.logger.WithField("error", err).Error("could not iterate over objects")
+		}
+
+		// Flush the propLengths to disk
+		err = index.IterateShards(ctx, func(index *Index, shard *Shard) error {
+			return shard.propLengths.Flush(false)
+		})
+		if err != nil {
+			m.logger.WithField("error", err).Error("could not flush prop lengths")
+		}
+
+	}
+	go func() {
```

Comment:
not the biggest fan of this, but I don't have a better idea

Reply:
I talked with Etienne about this for vector re-index and he wants something written to the logs.

Reply:
right now this routine seems to never finish. Shouldn't it be stopped at least at server shutdown? for example it could check the context canceled on shutdown

Reply:
or add a finalizer? LIke in the go client: https://github.com/weaviate/weaviate-go-client/blob/56a0292ff1d4388e6a887d0cc04bc48ec063b987/weaviate/connection/rest.go#L29